Source/SwiftLintFramework/Rules/Idiomatic/DuplicateImportsRule.swift

```diff
@@ -101,11 +193,23 @@ private extension Line {
         return self.content.split(separator: " ").last
     }
 
-    func areImportsDuplicated(with otherLine: Line) -> Bool {
-        guard let firstImportIdentifiers = self.importIdentifier?.split(separator: "."),
-            let secondImportIdentifiers = otherLine.importIdentifier?.split(separator: ".")
-            else { return false }
-
-        return zip(firstImportIdentifiers, secondImportIdentifiers).allSatisfy { $0 == $1 }
+    /// For "import A.B.C" returns slices [["A", "B", "C"], ["A", "B"], ["A"]]
+    var importSlices: [ImportSlice] {
+        guard let importIdentifier = importIdentifier else { return [] }
+
+        let importedSubpathParts = importIdentifier.split(separator: ".").map { String($0) }
+        guard !importedSubpathParts.isEmpty else { return [] }
+
+        return [
+            ImportSlice(
+                subpath: importedSubpathParts[0..<importedSubpathParts.count],
+                type: .complete
+            )
+        ] + stride(from: 1, to: importedSubpathParts.count, by: 1).map {
```

Comment:
`stride` might be a bit over-engineered here. What about a simple range expression to iterate over?
```suggestion
        ] + (1 ..< importedSubpathParts.count).map {
```

Reply:
I agree. Replaced `stride` with a range expression